source/fab/source_tree.py

```diff
@@ -33,40 +34,35 @@ def __init__(self,
                  command_flags_map: Mapping[Type[Command], List[str]],
                  state: SqliteStateDatabase,
                  workspace: Path,
-                 queue: QueueManager):
+                 task_handler: Callable):
         self._extension_map = extension_map
         self._command_flags_map = command_flags_map
         self._state = state
         self._workspace = workspace
-        self._queue = queue
+        self._task_handler = task_handler
 
     def visit(self, candidate: Path) -> List[Path]:
         new_candidates: List[Path] = []
         try:
             task_class = self._extension_map[candidate.suffix]
             reader: TextReader = FileTextReader(candidate)
-            hasher: TextReaderAdler32 = TextReaderAdler32(reader)
 
             if issubclass(task_class, Analyser):
-                task: Task = task_class(hasher, self._state)
+                task: Task = task_class(reader, self._state)
             elif issubclass(task_class, SingleFileCommand):
                 flags = self._command_flags_map.get(task_class, [])
                 task = CommandTask(
-                    task_class(Path(hasher.filename), self._workspace, flags))
+                    task_class(Path(reader.filename), self._workspace, flags))
             else:
                 message = \
                     f'Unhandled class "{task_class}" in extension map.'
                 raise TypeError(message)
-            # TODO: Make SQLite connection multiprocess safe
-            # self._queue.add_to_queue(task)
-            task.run()
+
+            self._task_handler(task)
+            self._task_handler(HashCalculator(reader, self._state))
```

Comment:
I don't think we should be hard wiring tasks into the descender. It should probably be for the layer above to decide whether files need to be hashed or not.

Reply:
After discussion with the developer we have agreed to go ahead with the change as it stands and reconsider the whole issue of `ExtensionVisitor` in ticket #58.

Reply:
Making this change is postponed to #58